server/logger.js

```diff
@@ -1,40 +1,17 @@
-const winston = require('winston');
-const fs = require('bfile');
+const Logger = require('blgr');
 
-/** Configure the logger */
-const logDir = __dirname.concat('/../logs');
-const env = process.env.NODE_ENV || 'development';
+const loadConfig = require('./utils/loadConfig');
 
-if (!fs.existsSync(logDir)) {
-  fs.mkdirSync(logDir);
-}
+const config = loadConfig('bpanel');
 
-const logger = new winston.Logger({
-  transports: [
-    new winston.transports.File({
-      level: env === 'development' ? 'debug' : 'info',
-      filename: logDir.concat('/logs.log'),
-      handleExceptions: true,
-      json: true,
-      maxsize: 5242880, // 5MB
-      maxFiles: 5,
-      colorize: false
-    }),
-    new winston.transports.Console({
-      level: 'debug',
-      handleExceptions: true,
-      json: false,
-      colorize: true
-    })
-  ],
-  exitOnError: false
+const logger = new Logger();
+logger.set({
+  filename: config.bool('log-file') ? config.location('debug.log') : null,
+  level: config.str('log-level', 'debug'),
+  console: config.bool('log-console', true),
+  shrink: config.bool('log-shrink', true)
 });
 
-/*
-  Stream is used to allow morgan to style console output
-  in server.js
-*/
+logger.open();
```

Comment:
Calling this without `async` does create a race condition, but it seems as if we are moving away from using this logger right?

Reply:
Correct. The main `open` is now done when it's attached to the `bpanelConfig` in server/index.js. and this is only called in cases where `bpanelConfig` is not available.